docs/reference/run.md

```diff
@@ -94,6 +94,20 @@ You can reattach to a detached container with `docker`
 [*attach*](/reference/commandline/attach). If you choose to run a
 container in the detached mode, then you cannot use the `--rm` option.
 
+When starting a container in detached mode, one needs to understand that, by
```

Comment:
@dharmit Thank you for the contribution.  A few things have slightly awkward constructions...also it isn't necessary to mention that `nginx` needs to be installed.  At this level of doc, that is obvious.

I like your additional detail on detached containers...needs to be at the start of this section.  So, let's expand and clarify.

---> https://gist.github.com/moxiegirl/5766ad333ed56ed7f9d7

### Detached (-d)

To start a container in detached mode, you use `-d=true` or just `-d` option. By design, containers started in detached mode exit when the root process used to run the container exits. A container is detached mode cannot be automatically removed when it stops, this means you cannot use the `--rm` option with `-d` option.

Do not pass a `service x start` command to a detached container. For example, this command attempts to start the `nginx` service.

```
$ docker run -d -p 80:80 my_image service nginx start
```

This attempt fails because the root process (`service x start`) returns and the detached container stops as designed. Instead, to start a process  `nginx` web server do the following:

    $ docker run -d -p 80:80 my_image nginx -g 'daemon off;

To do input/output with a detached container use network connections or shared
volumes. These are required because the container is no longer listening to the
command line where `docker run` was run. 

To reattach to a detached container, use with `docker`
[*attach*](/reference/commandline/attach) command.